### PR TITLE
Update accessions from rabbit messages.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #1201

# Why was this change made? 🤔

# How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (e.g. changes what is written to shared file systems or how it uses APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: app/jobs/job_run_complete_job.rb
Comment (justinlittman) [inline app/jobs/job_run_complete_job.rb:6]: Will be implementing this in separate PR.

**Review by ndushay — APPROVED** [review]
i've only skimmed it quickly, so would be good if someone looked more closely

File: db/migrate/20230908185452_add_state_to_accession.rb
Comment (jmartin-sul) [inline db/migrate/20230908185452_add_state_to_accession.rb:3]: is the reason for the default that an `accessions` row only gets created once the accessioning is launched for the druid?

File: app/jobs/accession_complete_job.rb
Comment (mjgiarlo) [inline app/jobs/accession_complete_job.rb:45]: I wonder if this might work too:

```suggestion
    accessions.update_all(state:)
```

Though the behavior on the batch update is different, as 
> It does not instantiate the involved models and it does not trigger Active Record callbacks or validation

**Review by mjgiarlo — APPROVED** [review]
Approve pending consideration of comment(s) which may or may not be worth doing

File: app/jobs/accession_complete_job.rb
Comment (jmartin-sul) [inline app/jobs/accession_complete_job.rb:11]: question due to my rabbit naivete: is the `from_queue 'preassembly.accession_complete'` enough to get this class to also receive `accession end-accession.completed`, due to some other rabbit config?  or is that wiring to come later?  something to do with the `queue.bind` in the `rabbit.rake`?  i've never been clear on what exactly topics and queues represent in rabbit (happy to go do some reading if this should be obvious).

File: app/models/accession.rb
Comment (jmartin-sul) [inline app/models/accession.rb:7]: cool, didn't know you could declare an `enum` like this!

File: spec/jobs/accession_complete_job_spec.rb
Comment (jmartin-sul) [inline spec/jobs/accession_complete_job_spec.rb:43]: super nitpicky and would not hold up the PR over this, but my understanding is that we should never have two versions of the same druid be in progress in accessioning at the same time.  if that understanding is correct, i'd lobby mildly for changing this example to... something else, i dunno what, though.  starts and stays `completed`, with no update timestamps? 🤷

**Review by jmartin-sul — APPROVED** [review]
some questions and a very small possible nitpick, but LGTM overall

File: app/jobs/accession_complete_job.rb
Comment (mjgiarlo) [reply app/jobs/accession_complete_job.rb:11]: My head hasn't been it as recently as @justinlittman's, but, short answer: yes. Slightly longer answer is `from_queue` here tells sneakers which queue to connect to. The value matches what was set up on line 17 in the setup task:

https://github.com/sul-dlss/pre-assembly/pull/1247/files#diff-cfb3e45018fb1025f9ed2356abd6d3f9f58808e7ef269f9c1bf1380e9d9283a9R17

And then that queue was bound go the `sdr.workflow` topic using two routing keys (`end-accession.completed` and any errors) here:

https://github.com/sul-dlss/pre-assembly/pull/1247/files#diff-cfb3e45018fb1025f9ed2356abd6d3f9f58808e7ef269f9c1bf1380e9d9283a9R18-R19

The first five tutorials here cover this in greater detail:

https://www.rabbitmq.com/getstarted.html

File: app/jobs/accession_complete_job.rb
Comment (justinlittman) [reply app/jobs/accession_complete_job.rb:45]: It would, but that would bypass the AR enum handling.

File: db/migrate/20230908185452_add_state_to_accession.rb
Comment (justinlittman) [reply db/migrate/20230908185452_add_state_to_accession.rb:3]: Correct.

File: app/jobs/accession_complete_job.rb
Comment (justinlittman) [reply app/jobs/accession_complete_job.rb:11]: See the creation of the rabbit queue in `rabbit.rake`.

File: app/models/accession.rb
Comment (justinlittman) [reply app/models/accession.rb:7]: Yes, I greatly dislike numerical enums because they make it more difficult to work directly with the DB.

Comment (jmartin-sul) [reply app/models/accession.rb:7]: for sure, same

File: app/jobs/accession_complete_job.rb
Comment (jmartin-sul) [reply app/jobs/accession_complete_job.rb:11]: thanks for the explanations and the doc link!

File: spec/jobs/accession_complete_job_spec.rb
Comment (justinlittman) [reply spec/jobs/accession_complete_job_spec.rb:43]: I added a comment indicating that this test was cutting a corner to avoid confusing our future selves.

Comment (jmartin-sul) [reply spec/jobs/accession_complete_job_spec.rb:43]: merged, thanks!

**Pull request merged**
